modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/VerticalNavTag.java

```diff
@@ -0,0 +1,229 @@
+ * details.
+ */
+
+package com.liferay.frontend.taglib.clay.servlet.taglib;
+
+import com.liferay.frontend.taglib.clay.internal.servlet.taglib.BaseContainerTag;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.VerticalNavItem;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.VerticalNavItemList;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.jsp.JspException;
+import javax.servlet.jsp.JspWriter;
+
+/**
+ * @author Eduardo Allegrini
+ * @author Daniel Sanz
+ */
+public class VerticalNavTag extends BaseContainerTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		setContainerElement("nav");
+
+		return super.doStartTag();
+	}
+
+	public boolean getDecorated() {
+		return _decorated;
+	}
+
+	public boolean getLarge() {
+		return _large;
+	}
+
+	public List<VerticalNavItem> getVerticalNavItems() {
+		return _verticalNavItems;
+	}
+
+	public void setDecorated(boolean decorated) {
+		_decorated = decorated;
+	}
+
+	public void setLarge(boolean large) {
+		_large = large;
+	}
+
+	public void setVerticalNavItems(List<VerticalNavItem> verticalNavItems) {
+		_verticalNavItems = verticalNavItems;
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_decorated = false;
+		_large = false;
+		_verticalNavItems = null;
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(Map<String, Object> props) {
+		props.put("decorated", _decorated);
+		props.put("large", _large);
+		props.put("items", _verticalNavItems);
+
+		return super.prepareProps(props);
+	}
+
+	@Override
+	protected String processCssClasses(Set<String> cssClasses) {
+		cssClasses.add("menubar menubar-transparent");
+
+		if (_decorated) {
+			cssClasses.add("menubar-decorated");
+		}
+
+		cssClasses.add(
+			_large ? "menubar-vertical-expand-lg" :
+				"menubar-vertical-expand-md");
+
+		return super.processCssClasses(cssClasses);
+	}
+
+	@Override
+	protected int processStartTag() throws Exception {
+		super.processStartTag();
+
+		JspWriter jspWriter = pageContext.getOut();
+
+		jspWriter.write("<div class=\"collapse menubar-collapse\">");
+
+		_renderVerticalNavItems(jspWriter, _verticalNavItems, 0);
+
+		jspWriter.write("</div>");
+
+		return EVAL_BODY_INCLUDE;
+	}
+
+	private void _renderVerticalNavItems(
+			JspWriter jspWriter, List<VerticalNavItem> verticalNavItems,
+			int depth)
+		throws Exception {
+
+		jspWriter.write("<ul aria-orientation=\"vertical\" role=\"menubar\"");
+
+		jspWriter.write("class=\"nav ");
+
+		if (depth == 0) {
+			jspWriter.write("nav-nested\">");
+		}
+		else {
+			jspWriter.write("nav-stacked\">");
+		}
+
+		for (VerticalNavItem verticalNavItem : verticalNavItems) {
+			VerticalNavItemList items =
+				(VerticalNavItemList)verticalNavItem.get("items");
+
+			Boolean active = (Boolean)verticalNavItem.get("active");
```

Comment:
Why this complexity with unsafe suppliers and castings? Why can't `VerticalNavItem` be a POJO with simple getters, `isActive` in this case?

Reply:
Here I've followed the pattern for NavigationItem family of classes. We can consider this in a follow up, my thoughts at this moment are:
* NavigationItem and friends are wrappers on top of the `map` api, so not strictly needed anyway other than for typing purposes
* Setters are part of the public API for whoever uses the tag from DXP. So they add meaning and purpose to client code.
* However, the getters (which don't exist at this moment) are meant to be used internally, i.e. from the tag implementations.
* Adding the getters might make client code believe they're there for some reason, but they're useless from the tag usage pov

Perhaps the last 2 points made implementors decide not to add the getters.